src/animalShelter/AnimalShelter.ts

```diff
@@ -16,93 +29,98 @@ export default class AnimalShelter {
   }
-  setMammalLimit(newMammalLimit: number) {
+  setMammalLimit(newMammalLimit: number): string {
     if (newMammalLimit < 0) {
-      console.log(`Limit cannot be negative`);
-      return;
+      return NEGATIVE_LIMIT_ERROR_MESSAGE;
     }
     const mamalsTotal = this.getMammalsCount();
     if (newMammalLimit >= mamalsTotal) {
       this._mammalLimit = newMammalLimit;
-      console.log("New Mammal Limit is now : " + this._mammalLimit);
+      return SUCCESFULLY_SET_NEW_MAMMAL_LIMIT;
     } else {
-      console.log(
-        `Cannot change mammal limit to ${newMammalLimit} there are already ${mamalsTotal} mammals`
-      );
+      return NEW_MAMMAL_LIMIT_LESS_THAN_CURRENT_MAMMALS;
     }
   }
-
   private getReptileCount() {
-    return this.animals.filter(x => x instanceof Reptile).length;
+    return this.animals.filter((x) => x instanceof Reptile).length;
   }
   getReptileLimit() {
     return this._reptileLimit;
   }
-  setReptileLimit(newReptileLimit: number) {
+  setReptileLimit(newReptileLimit: number): string {
     if (newReptileLimit < 0) {
-      console.log(`Limit cannot be negative`);
-      return;
+      return NEGATIVE_LIMIT_ERROR_MESSAGE;
     }
     const reptilesTotal = this.getReptileCount();
     if (newReptileLimit >= reptilesTotal) {
       this._reptileLimit = newReptileLimit;
-      console.log("New Reptile Limit is now : " + this._reptileLimit);
+      return SUCCESFULLY_SET_NEW_REPTILE_LIMIT;
     } else {
-      console.log(
-        `Cannot change reptile limit to ${newReptileLimit} there ${reptilesTotal == 1 ? "is" : "are"} already ${reptilesTotal} ${reptilesTotal == 1 ? "reptile" : "reptile"}`
-      );
+      return NEW_REPTILE_LIMIT_LESS_THAN_CURRENT_REPTILES;
     }
   }
 
-
   showAnimals() {
-
-    console.log(this.animals);
-    console.log("mammals: " + this.getMammalsCount());
-    console.log("reptiles: " + this.getReptileCount());
-    console.log(`lions: ${this.animals.filter(lion => lion instanceof Lion).length}`);
-    console.log(`crocodiles: ${this.animals.filter(crocodile => crocodile instanceof Crocodile).length}`);
-    console.log(`squirrels: ${this.animals.filter(squirrel => squirrel instanceof Squirrel).length}`);
-    console.log(`snakes: ${this.animals.filter(snake => snake instanceof Snake).length}`);
-    console.log(`tortoises: ${this.animals.filter(tortoise => tortoise instanceof Tortoise).length}`);
+    const animalNumbers: {
+      mamamls: number;
+      reptiles: number;
+      lions: number;
+      crocodiles: number;
+      squirrels: number;
+      snakes: number;
+      tortoises: number;
+    } = {
+      mamamls: this.getMammalsCount(),
+      reptiles: this.getReptileCount(),
+      lions: this.animals.filter((lion) => lion instanceof Lion).length,
+      crocodiles: this.animals.filter(
+        (crocodile) => crocodile instanceof Crocodile
+      ).length,
+      squirrels: this.animals.filter((squirrel) => squirrel instanceof Squirrel)
```

Comment:
not a blocker but when naming the animal inside the filter , having a "squirrel" instead of animal can be deceiving in a bit more complex code. Here "animal" is more appropriate for a name